raiden/network/proxies/user_deposit.py

```diff
@@ -360,6 +386,61 @@ def approve_and_deposit(
 
             self._deposit_check_result(transaction_sent, token, total_deposit, amount_to_deposit)
 
+    def plan_withdraw(
+        self, amount: TokenAmount, given_block_identifier: BlockIdentifier
+    ) -> BlockNumber:
+        """ Announce that you plan to withdraw tokens from the UserDeposit contract
+
+        Returns the block number at which the withdraw is ready.
+        """
+        self._plan_withdraw_preconditions(amount, given_block_identifier)
+
+        # Simplify our lives by disallowing concurrent plan_withdraw / withdraw calls
+        with self._withdraw_lock:
+            estimated_transaction = self.client.estimate_gas(
+                self.proxy, "planWithdraw", {}, amount
+            )
+            transaction_sent = None
+            if estimated_transaction is not None:
+                transaction_sent = self.client.transact(estimated_transaction)
+            transaction_mined = self._plan_withdraw_check_result(
+                transaction_sent=transaction_sent, amount_to_plan_withdraw=amount
+            )
+
+        assert transaction_mined is not None, "_plan_withdraw_check_result returned None"
```

Comment:
~this assert seems incorrect, if gas estimation fails the transaction is not sent, therefore it cannot be mined.~ This is wrong, `_plan_withdraw_check_result` always raises if `transaction_sent` is `None`. I'd say the split of checks into separate methods made the control flow unnecessarily hard convoluted, maybe invert the check move at line `404` and raise there?

Reply:
Yes I agree that this is not very nice. I thought it better to keep with the already existing style in the module though.

This assert is indeed useless. Initially `_plan_withdraw_check_result` didn't always raise. But that was even harder to understand.